packages/flutter/test/material/page_transitions_theme_test.dart

```diff
@@ -235,6 +235,40 @@ void main() {
     expect(coloredBox.color, Colors.transparent); // Color is transparent during animation.
   }, variant: TargetPlatformVariant.only(TargetPlatform.android));
 
+  testWidgets('CupertinoPageTransitionsBuilder default duration is 500ms', (WidgetTester tester) async {
+    final Map<String, WidgetBuilder> routes = <String, WidgetBuilder>{
+      '/': (BuildContext context) => Material(
+        child: TextButton(
+          child: const Text('push'),
+          onPressed: () { Navigator.of(context).pushNamed('/b'); },
+        ),
+      ),
+      '/b': (BuildContext context) => const Text('page b'),
+    };
+
+    await tester.pumpWidget(
+      MaterialApp(
+        theme: ThemeData(
+          pageTransitionsTheme: const PageTransitionsTheme(
+            builders: <TargetPlatform, PageTransitionsBuilder>{
+              TargetPlatform.iOS: CupertinoPageTransitionsBuilder()
+            },
+          ),
+        ),
+        routes: routes,
+      ),
+    );
+
+    expect(find.byType(CupertinoPageTransition), findsOneWidget);
+
+    await tester.tap(find.text('push'));
+    await tester.pump(const Duration(milliseconds: 499));
+    expect(tester.hasRunningAnimations, isTrue);
+
+    await tester.pump(const Duration(milliseconds: 501));
```

Comment:
Doing a pump for 499 milliseconds then another one for 501 feels odd, when we are testing for 500. It feels like logically a wide range times could also pass this test.

I haven't ran this locally, but I am assuming that doing a shorter duration for the second pump doesn't allow the test to pass? When we are testing certain points of a page transition, we often have to do a pump first to trigger the start of a transition, then do another pump with the correct duration.

So I'm wondering if this would work and be closer to what we want to test:
```
await tester.pump();
// Transition has been triggered but not begun yet
await tester.pump(const Duration(milliseconds: 499));
// check if animating
await tester.pump(const Duration(milliseconds: 50)); -> or any small duration
// check if finished
```

Reply:
Thank you for pointing that out! I've made the suggested changes (09b9650).